prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -72,7 +73,19 @@ ssize_t rxr_pkt_post_data(struct rxr_ep *rxr_ep,
 	 */
 	data_pkt->hdr.seg_offset = tx_entry->bytes_sent;
 
+	/*
+	 * TODO: Check to see if underlying device can support CUDA
+	 * registrations and fallback to rxr_ep_send_data_pkt_entry() if it does
+	 * not. This should be done at init time with a CUDA reg-and-fail flag.
+	 * For now, always send CUDA buffers through
+	 * rxr_ep_mr_send_data_pkt_entry().
+	 */
+#ifdef HAVE_LIBCUDA
+	if (efa_mr_cache_enable ||
+	    rxr_ep_is_cuda_mr(tx_entry->desc[0]))
+#else
 	if (efa_mr_cache_enable)
+#endif
 		ret = rxr_pkt_send_data_mr_cache(rxr_ep, tx_entry, pkt_entry);
```

Comment:
Perhaps rename this function to something more generic now that there's another avenue where a descriptor is being used?